projects/components/src/datagrid/datagrid.component.spec.ts

```diff
@@ -243,6 +242,13 @@ describe('DatagridComponent', () => {
                     this.clrGridWidget.selectRow(1);
                     expect(this.finder.hostComponent.datagridSelection).toEqual([mockData[1]]);
                 });
+
+                it('reads an empty array and returns an empty array', function (this: HasFinderAndGrid): void {
+                    this.finder.hostComponent.selectionType = GridSelectionType.Single;
+                    this.finder.detectChanges();
+                    this.finder.hostComponent.datagridSelection = [];
+                    expect(this.finder.hostComponent.datagridSelection).toEqual([]);
```

Comment:
Unless you run change detection between line 249 and 250, this test isn't asserting anything interesting. If you run change detection between these lines, it should allow your setter on line 249 to propagate to the datagrid component and for it to update the value accordingly. (reach out to me if you want any help with how change detection works, not that I'm an expert at all hahahaha)

Reply:
added this.finder.detectChanges(), ran tests again, all pass